File: doc-examples/hibernate-example-java/src/main/java/example/Student.java

package example;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Version;

// tag::student[]
@Entity
public class Student {

    @Id
    @GeneratedValue
    private Long id;
    @Version
    private Long version;

    // end::student[]

    private String name;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getVersion() {
        return version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: doc-examples/hibernate-reactive-example-kotlin/src/main/kotlin/example/Student.kt
package example

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Version

// tag::student[]
@Entity
data class Student(
        @Id @GeneratedValue
        var id: Long?,
        @Version
        val version: Long,

        // end::student[]

        var name: String
) {
    constructor(name: String) : this(null, 0, name)
}
